File: oapp/services/menu/internal/models/collection.go
package models

import "time"

// Collection represents a menu collection (category)
type Collection struct {
	CollectionID string    `json:"collection_id" db:"collection_id"`
	MenuID       string    `json:"menu_id" db:"menu_id"`
	Name         string    `json:"name" db:"name"`
	Position     int       `json:"position" db:"position"`
	ImageURL     *string   `json:"image_url,omitempty" db:"image_url"`
	CreatedAt    time.Time `json:"created_at" db:"created_at"`
	UpdatedAt    time.Time `json:"updated_at" db:"updated_at"`
}

// CreateCollectionRequest represents the request to create a collection
type CreateCollectionRequest struct {
	Name     string  `json:"name" validate:"required,oneof=Pizzas Burgers Sides Desserts Drinks Salads Breakfast Sandwiches Pasta Seafood"`
	Position int     `json:"position" validate:"min=0"`
	ImageURL *string `json:"image_url,omitempty"`
}

// ReorderCollectionsRequest represents the request to reorder collections
type ReorderCollectionsRequest struct {
	Collections []CollectionPosition `json:"collections" validate:"required,dive"`
}

// CollectionPosition represents a collection ID and its new position
type CollectionPosition struct {
	CollectionID string `json:"collection_id" validate:"required"`
	Position     int    `json:"position" validate:"min=0"`
}

File: oapp/DeliveryXOApp/DeliveryXOApp/Models/Collection.swift
//
//  Collection.swift
//  DeliveryXOApp
//
//  Created on 2025-10-21.
//

import Foundation

/// Predefined collection names/categories
enum CollectionName: String, Codable, CaseIterable {
    case pizzas = "Pizzas"
    case burgers = "Burgers"
    case sides = "Sides"
    case desserts = "Desserts"
    case drinks = "Drinks"
    case salads = "Salads"
    case breakfast = "Breakfast"
    case sandwiches = "Sandwiches"
    case pasta = "Pasta"
    case seafood = "Seafood"
}

/// Represents a collection/category in a menu
struct Collection: Codable, Identifiable {
    let id: String
    let menuId: String
    let name: CollectionName
    let position: Int?
    let imageUrl: String?
    let createdAt: Date
    let updatedAt: Date

    enum CodingKeys: String, CodingKey {
        case id = "collection_id"
        case menuId = "menu_id"
        case name
        case position
        case imageUrl = "image_url"
        case createdAt = "created_at"
        case updatedAt = "updated_at"
    }
}

/// Collection with its products included
struct CollectionWithProducts: Codable, Identifiable {
    let id: String
    let menuId: String
    let name: CollectionName
    let position: Int?
    let imageUrl: String?
    let createdAt: Date
    let updatedAt: Date
    let products: [ProductInMenu]

    enum CodingKeys: String, CodingKey {
        case id = "collection_id"
        case menuId = "menu_id"
        case name
        case position
        case imageUrl = "image_url"
        case createdAt = "created_at"
        case updatedAt = "updated_at"
        case products
    }
}

/// Request model for creating a new collection
struct CreateCollectionRequest: Codable {
    let name: CollectionName
    let position: Int?
    let imageUrl: String?

    enum CodingKeys: String, CodingKey {
        case name
        case position
        case imageUrl = "image_url"
    }
}
